include/oneapi/tbb/detail/_template_helpers.h

```diff
@@ -182,7 +182,17 @@ using pack_element_t = typename pack_element<N, Args...>::type;
 template <typename Func>
 class raii_guard {
 public:
-    raii_guard( Func f ) : my_func(f), is_active(true) {}
+    static_assert(
+        std::is_nothrow_copy_constructible<Func>::value &&
+        std::is_nothrow_move_constructible<Func>::value,
+        "Throwing an exception during the Func copy or move construction cause an unexpected behavior."
+    );
+
+    raii_guard( Func f ) noexcept : my_func(f), is_active(true) {}
+
+    raii_guard( raii_guard&& g ) noexcept : my_func(std::move(g.my_func)), is_active(g.is_active) {
```

Comment:
please be so kind to add a test for it

Reply:
What cases do you want to test? 

Getting `std::terminate` in the case when the functor copy/move constructor is noexcept but throws an exception? Isn't the termination, in this case, guaranteed by the C++ standard?

Reply:
simple case that given (`noexcept`) function is called once (in appropriate time), if the one `raii_guard` instance is move constricted from the other one

Reply:
As I know the common approach here is do not to create a separate test for internal functionality if it already covered by public interfaces testing. If I understand your case correctly, then it already well covered by concurrent container testing.